File: tests/.hooks/pythonv2/secondary/src/openapi/_hooks/customsecurityhook.py
from typing import Union, cast
import httpx

from openapi.models.operations.customschemeappid import CustomSchemeAppIDSecurity

from .types import (
    BeforeRequestContext,
    BeforeRequestHook,
)


class CustomSecurityHook(BeforeRequestHook):
    def before_request(
        self, hook_ctx: BeforeRequestContext, request: httpx.Request
    ) -> Union[httpx.Request, Exception]:
        if hook_ctx.operation_id == "customSchemeAppId":
            if hook_ctx.security_source is None:
                return Exception("security source is None")

            sec = hook_ctx.security_source
            if callable(sec):
                sec = sec()

            custom_sec = cast(CustomSchemeAppIDSecurity, sec)
            request.headers["X-Security-App-Id"] = custom_sec.app_id
            request.headers["X-Security-Secret"] = custom_sec.secret

        return request

File: tests/.hooks/typescriptv2/secondary/src/hooks/customsecurityhook.ts
import { BeforeRequestContext, BeforeRequestHook } from "./types.js";

import { CustomSchemeAppIdSecurity } from "../sdk/models/operations/customschemeappid.js";

export class CustomSecurityHook implements BeforeRequestHook {
  beforeRequest(hookCtx: BeforeRequestContext, request: Request): Request {
    switch (hookCtx.operationID) {
      case "customSchemeAppId": {
        let sec = hookCtx.securitySource;
        if (typeof sec === "function") {
          sec = sec();
        }
        if (!sec) {
          throw new Error("security source is not defined");
        }

        const customSec = sec as CustomSchemeAppIdSecurity;
        request.headers.set("X-Security-App-Id", customSec.appId);
        request.headers.set("X-Security-Secret", customSec.secret);

        break;
      }
    }

    return request;
  }
}
